crates/matrix-sdk-ui/src/timeline/builder.rs

```diff
@@ -158,9 +158,17 @@ impl TimelineBuilder {
         let (_, mut event_subscriber) = room_event_cache.subscribe().await?;
 
         let is_live = matches!(focus, TimelineFocus::Live);
-
-        let inner = TimelineInner::new(room, focus, internal_id_prefix, unable_to_decrypt_hook)
-            .with_settings(settings);
+        let is_room_encrypted =
+            room.is_encrypted().await.map_err(|_| Error::UnknownEncryptionState)?;
```

Comment:
I wonder if this is going to slow down the timeline or create weird problems if it fails. I imagine that at this point `is_encrypted` is already fetched and cached so I don't think so.

Reply:
Yeah this one should be fine from my testing.